### ISSUE TITLE
Segment/region detection

### ISSUE THREAD

**Issue by jbarth-ubhd** [OCR4all/OCR4all#90]
If ocrd-pc-segmenter is the same as the segmentation algorithm in ocr4all, this comparison might interesting to you: https://digi.ub.uni-heidelberg.de/diglitData/v/testset-ls-v3.pdf

**Comment by chreul** [comment]
we aren't using ocrd-pc-segment in ocr4all. actually, we are still looking for a robust fully automatic segmentation solution, so thanks for the comparison! the link seems to be down, though.

**Comment by jbarth-ubhd** [comment]
Kernel update. Is up now.

**Comment by jbarth-ubhd** [comment]
Is there no real region segmentation in ocr4all? `Segmentation (Dummy)`

**Comment by chreul** [comment]
at the start of the project the main focus of ocr4all lied on the high quality processing of very early printings like incunabula where a quite considerable amount of human effort is often perfectly acceptable. consequently, we focused on developing the interactive solution (LAREX) first.
later and due to popular demand we added a fully automatic solution. as a stop-gap we use the ocropus segmentation which, apart from the column detection, has at least some text/non-text separation capabilities. currently, this is deferred to the line segmentation step and to keep the workflow consistent with the initial one we introduced the dummy region segmentation.
naturally, this is not intended to be a long-term solution and we are looking to add a selection of the current OCR-D processors as options for this step.

**Comment by jbarth-ubhd** [comment]
I'm processing 72 images and ocr4all `pagelineseg` is running for 154 minutes ... just now finished. Almost 1 core used, RSS approx max. 4 GB. 

calamari-predict seems to use all CPUs. Perhaps you could add parallelization of `pagelineseg`.

PS: cleared my browser cache, cookies etc. How do I get my currently running project?

**Comment by jbarth-ubhd** [comment]
Got results in `processing/`. 

Unsurprisingly your line segmenter does not always recognize columns with separating line:

```xml
<Unicode>Wie die &#x201E;L. N. melden, hat &#x17F;ich am Breslau ver&#x17F;ucht. Hierbei 
wurden n&#xE4;mlich</Unicode>
```
![grafik](https://user-images.githubusercontent.com/30653779/96451906-8d0ebb00-1218-11eb-9dd1-3c2d4299ecef.png)

**Comment by jbarth-ubhd** [comment]
On the positive side it seldom misses text. Counterexamples (complete testset: 72 pages): 

![grafik](https://user-images.githubusercontent.com/30653779/96454530-4753f180-121c-11eb-88d9-9feba3e5ac43.png)

**Comment by maxnth** [comment]
>ocr4all pagelineseg is running for 154 minutes ... just now finished

Lineseg has the tendency to run quite long for certain pages. This is most likely caused by the smearing algorithm we use and the standard parameters we set. We're looking to fix that in one of the upcoming OCR4all releases.

>Perhaps you could add parallelization of pagelineseg

IIRC pagelineseg should run in parallel but I'll look into it.

>PS: cleared my browser cache, cookies etc. How do I get my currently running project?

There's a bug with the current session management which causes processes to fail/freeze in the UI. Selecting `Project Overview` and then the workflow step corresponding to the process in the sidebar "fixes" this. Admittedly this is a bit messy but we're already planning to rewrite workflow handling with the newly implemented REST API which also allows more fine grained process scheduling.
